core/templates/pages/exploration-player-page/new-lesson-player/new-lesson-player-components/player-sidebar.component.ts

```diff
@@ -16,21 +16,86 @@
  * @fileoverview Component for the new lesson player sidebar
  */
 
-import { Component } from '@angular/core';
+import { Component, OnInit } from '@angular/core';
 import { downgradeComponent } from '@angular/upgrade/static';
+import { MobileMenuService } from '../new-lesson-player-services/mobile-menu.service';
 import './player-sidebar.component.css';
+import { ContextService } from 'services/context.service';
+import { I18nLanguageCodeService, TranslationKeyType } from
+  'services/i18n-language-code.service';
+import { ReadOnlyExplorationBackendApiService } from 'domain/exploration/read-only-exploration-backend-api.service';
+import { UrlService } from 'services/contextual/url.service';
 
 @Component({
   selector: 'oppia-player-sidebar',
   templateUrl: './player-sidebar.component.html',
   styleUrls: ['./player-sidebar.component.css'],
 })
-export class PlayerSidebarComponent {
-  isExpanded = false;
+export class PlayerSidebarComponent implements OnInit {
+  mobileMenuVisible: boolean = false;
+  isExpanded: boolean = false;
+  explorationId!: string;
+  expDesc!: string;
+  expDescTranslationKey!: string;
+  avgRating!: number | null;
+  fullStars: number = 0;
+  blankStars: number = 5;
+
+  constructor(
+    private mobileMenuService: MobileMenuService,
+    private contextService: ContextService,
+    private i18nLanguageCodeService: I18nLanguageCodeService,
+    private readOnlyExplorationBackendApiService:
+    ReadOnlyExplorationBackendApiService,
+    private urlService: UrlService,
+  ) {}
+
+  ngOnInit(): void {
+    this.mobileMenuService.getMenuVisibility().subscribe((visibility) => {
+      this.mobileMenuVisible = visibility;
+    });
+    let pathnameArray = this.urlService.getPathname().split('/');
+    let explorationContext = false;
+
+    for (let i = 0; i < pathnameArray.length; i++) {
+      if (pathnameArray[i] === 'explore' ||
+          pathnameArray[i] === 'create' ||
+          pathnameArray[i] === 'skill_editor' ||
+          pathnameArray[i] === 'embed' ||
+          pathnameArray[i] === 'lesson') {
+        explorationContext = true;
+        break;
+      }
+    }
+
+    this.explorationId = explorationContext ?
+      this.contextService.getExplorationId() : 'test_id';
+    this.expDesc = 'Loading...';
```

Comment:
Try to use full words, don't abbreviate. Otherwise it leads to ambiguity (e.g. for expDescTranslationKey, "descending" was the first word I thought of).

Reply:
Done

Reply:
I think they shouldn't be unnecessarily long when the context explains the meaning properly. Like I could change expDesc to expDescription but something like explorationDescriptionTranslationKey would just be so long that it becomes unreadable and isn't necessary in my opinion, does that make sense? I'll try to improve my phrasing for more clarity though

Reply:
@Vir-8 I'm fine with expDescriptionTranslationKey but I do think that Desc is ambiguous (it suggests descending and "keys" are often used for sorting) so I think it is worth changing. Another option is expDescriptionI18nKey.

Shorter names are better when possible, but I would suggest, as a rule of thumb, prioritizing clarity/non-ambiguity before prioritizing length. The issue is that you, as the author, generally have more context than another developer who reads the code, so things that are more obvious to you are not going to be as obvious for others. So, if it helps with making the names non-confusable with other things, it's fine if the names are longer.